### PR TITLE
add: support for jax.numpy.where

### PR DESCRIPTION
**Pull request by H4ckOm**

### CODE REVIEW

**Comment by H4ckOm** [comment]
Close #13123

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 1]
```diff
@@ -337,3 +337,42 @@ def test_jax_numpy_sort_complex(
...
+    test_with_out=st.just(False),
+    inplace=st.just(False),
+)
+def test_jax_numpy_where(
+    *,
+    dtype_and_x,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:355]: all of these can be removed,
you can instead just add argument `test_flags`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 2]
```diff
@@ -337,3 +337,42 @@ def test_jax_numpy_sort_complex(
...
+    on_device,
+    fn_tree,
+    frontend,
+):
+    input_dtype, x = dtype_and_x
+    x = helpers.get_frontend_ret(frontend, f"{frontend}.array", x[0], dtype=input_dtype)
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:361]: instead of this you can just do `x= x[0]`
since `x[0]` is a numpy array and thats what our `test_frontend_function` accepts

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

requested changes on specific lines

thanks

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 3]
```diff
@@ -337,3 +337,42 @@ def test_jax_numpy_sort_complex(
...
         test_values=False,
     )
+
+
+# where
+@handle_frontend_test(
+    fn_tree="jax.numpy.where",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("numeric")),
+    test_with_out=st.just(False),
+    inplace=st.just(False),
```
[the remark this answers is not in the document]
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have removed it

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 1 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have removed it

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 4]
```diff
@@ -337,3 +337,42 @@ def test_jax_numpy_sort_complex(
...
+    condition = x > 0.5
+    x1 = x * 2
+    x2 = x * -2
+
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
```
[the remark this answers is not in the document]
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 2 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it

**Review by H4ckOm — COMMENTED** [review]
@sherry30 Thanks for the review. I have updated the test as per your comments. Seems like it is failing now, please can you check?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 5]
```diff
@@ -337,3 +337,33 @@ def test_jax_numpy_sort_complex(
...
+
+
+# where
+@handle_frontend_test(
+    fn_tree="jax.numpy.where",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("numeric")),
+    test_with_out=st.just(False),
+)
+def test_jax_numpy_where(
+    test_flags,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:349]: you weren't supposed to remove `dtype_and_x`,
you can add this argument back in and also make the arguments keyword only by adding `*` first.

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

requested change on specific line again, once you fix that you'll get errors with the implementation.

You're going to have to look into them. Let me know if you get stuck anywhere, I'd be happy to help 😄 

thanks

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by H4ckOm** [comment]
Hey @sherry30 I've updated PR but still, the intelligent tests are failing. I tried to fix it but did not get any success. Please can you help?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by H4ckOm** [comment]
@sherry30

**Comment by sherry30** [comment]
Hi @H4ckOm,
Sorry for the late reply, I've been busy with other tasks 😅,
I'll look into this today and get back to you.

Thank you for the patience

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 6]
```diff
@@ -337,3 +337,38 @@ def test_jax_numpy_sort_complex(
...
+    native_arrays=st.just(True),
+)
+def test_jax_numpy_where(
+    *,
+    dtype_and_x,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+    native_array
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:356]: remove this

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 7]
```diff
@@ -337,3 +337,38 @@ def test_jax_numpy_sort_complex(
...
+    x = x[0]
+    condition = x > 0.5
+    x1 = x * 2
+    x2 = x * -2
+
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        frontend=frontend,
+        native_array_flags=native_array,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:368]: remove this

File: ivy/functional/frontends/jax/numpy/searching_sorting.py [hunk 8]
```diff
@@ -150,3 +150,10 @@ def flatnonzero(a):
 @to_ivy_arrays_and_back
 def sort_complex(a):
     return ivy.sort(a)
+
+
+def where(condition, x=None, y=None):
```
Comment (sherry30) [inline ivy/functional/frontends/jax/numpy/searching_sorting.py:155]: add `@to_ivy_arrays_and_back`  above this

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

I've asked for changes on specific lines, this should get the basic test working.
You also need to add `size` and `fill_value` argument. You can see them here https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.where.html
Le tme know if you have any other questions

Thanks

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 9]
```diff
@@ -337,3 +337,38 @@ def test_jax_numpy_sort_complex(
         a=x[0],
         test_values=False,
     )
+
+
+# where
+@handle_frontend_test(
+    fn_tree="jax.numpy.where",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("numeric")),
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:345]: for the function `helpers.dtype_and_values` you should also add the argument.
`min_num_dims=1` so that it doesn't produce 0d arrays

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 10]
```diff
@@ -337,3 +337,38 @@ def test_jax_numpy_sort_complex(
...
+    native_array
+):
+    input_dtype, x = dtype_and_x
+    x = x[0]
+    condition = x > 0.5
+    x1 = x * 2
+    x2 = x * -2
+
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:365]: tis should be `["bool", input_dtype, input_dtype]` because thats what we're passing here

**Review by sherry30 — CHANGES_REQUESTED** [review]
noticed 2 more things

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 10 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 9 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 7 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py (hunk 6 above)
Comment (H4ckOm) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py]: Thanks, I have updated it!

**Review by H4ckOm — COMMENTED** [review]
.
